corehq/blobs/interface.py

```diff
@@ -0,0 +1,75 @@
+"""Filesystem database for large binary data objects (blobs)
+"""
+from __future__ import absolute_import
+
+import re
+from abc import ABCMeta, abstractmethod
+from uuid import uuid4
+
+from corehq.blobs import DEFAULT_BUCKET
+from corehq.blobs.exceptions import BadName
+
+SAFENAME = re.compile("^[a-z0-9_./-]+$", re.IGNORECASE)
+
+
+class AbstractBlobDB(object):
+    """Filesystem storage for large binary data objects
+    """
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def put(self, content, basename="", bucket=DEFAULT_BUCKET):
+        """Put a blob in persistent storage
+        :param content: A file-like object in binary read mode.
```

Comment:
[Convention](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings) is to put a blank line between the first (summary) line of a docstring and any following lines.